pkg/helm/upgrade_test.go

```diff
@@ -10,33 +11,98 @@ import (
 
+type UpgradeTest struct {
+	expectedCommand string
+	upgradeStep     UpgradeStep
+}
+
 func TestMixin_Upgrade(t *testing.T) {
-	os.Setenv(test.ExpectedCommandEnv, `helm upgrade MYRELEASE MYCHART --namespace MYNAMESPACE --version 1.0.0 --replace --values /tmp/val1.yaml --values /tmp/val2.yaml --set baz=qux --set foo=bar`)
-	defer os.Unsetenv(test.ExpectedCommandEnv)
-
-	step := UpgradeStep{
-		Arguments: UpgradeArguments{
-			Namespace: "MYNAMESPACE",
-			Name:      "MYRELEASE",
-			Chart:     "MYCHART",
-			Version:   "1.0.0",
-			Replace:   true,
-			Set: map[string]string{
-				"foo": "bar",
-				"baz": "qux",
+	namespace := "MYNAMESPACE"
+	name := "MYRELEASE"
+	chart := "MYCHART"
+	version := "1.0.0"
+	setArgs := map[string]string{
+		"foo": "bar",
+		"baz": "qux",
+	}
+	values := []string{
+		"/tmp/val1.yaml",
+		"/tmp/val2.yaml",
+	}
+
+	baseUpgrade := fmt.Sprintf(`helm upgrade %s %s --namespace %s --version %s`, name, chart, namespace, version)
+	baseValues := `--values /tmp/val1.yaml --values /tmp/val2.yaml`
+	baseSetArgs := `--set baz=qux --set foo=bar`
+
+	upgradeTests := []UpgradeTest{
+		UpgradeTest{
+			expectedCommand: fmt.Sprintf(`%s %s %s`, baseUpgrade, baseValues, baseSetArgs),
+			upgradeStep: UpgradeStep{
+				Arguments: UpgradeArguments{
+					Namespace: namespace,
+					Name:      name,
+					Chart:     chart,
+					Version:   version,
+					Set:       setArgs,
+					Values:    values,
+				},
+			},
+		},
+		UpgradeTest{
+			expectedCommand: fmt.Sprintf(`%s %s %s %s`, baseUpgrade, `--reset-values`, baseValues, baseSetArgs),
+			upgradeStep: UpgradeStep{
+				Arguments: UpgradeArguments{
+					Namespace:   namespace,
+					Name:        name,
+					Chart:       chart,
+					Version:     version,
+					Set:         setArgs,
+					Values:      values,
+					ResetValues: true,
+				},
+			},
+		},
+		UpgradeTest{
+			expectedCommand: fmt.Sprintf(`%s %s %s %s`, baseUpgrade, `--reuse-values`, baseValues, baseSetArgs),
+			upgradeStep: UpgradeStep{
+				Arguments: UpgradeArguments{
+					Namespace:   namespace,
+					Name:        name,
+					Chart:       chart,
+					Version:     version,
+					Set:         setArgs,
+					Values:      values,
+					ReuseValues: true,
+				},
 			},
-			Values: []string{
-				"/tmp/val1.yaml",
-				"/tmp/val2.yaml",
+		},
+		UpgradeTest{
+			expectedCommand: fmt.Sprintf(`%s %s %s %s`, baseUpgrade, `--wait`, baseValues, baseSetArgs),
+			upgradeStep: UpgradeStep{
+				Arguments: UpgradeArguments{
+					Namespace: namespace,
+					Name:      name,
+					Chart:     chart,
+					Version:   version,
+					Set:       setArgs,
+					Values:    values,
+					Wait:      true,
+				},
 			},
 		},
 	}
-	b, _ := yaml.Marshal(step)
 
-	h := NewTestMixin(t)
-	h.In = bytes.NewReader(b)
+	for _, upgradeTest := range upgradeTests {
+		os.Setenv(test.ExpectedCommandEnv, upgradeTest.expectedCommand)
+		defer os.Unsetenv(test.ExpectedCommandEnv)
 
-	err := h.Upgrade()
+		b, _ := yaml.Marshal(upgradeTest.upgradeStep)
```

Comment:
Is it possible that Marshall could fail at all? For the sanity of whoever manages to get this to fail in the future I recommend this:

```suggestion
		b, err := yaml.Marshal(upgradeTest.upgradeStep)
		require.NotNil(t, err)
```

Reply:
(oops, needed some slight adjustments, but just pushed via an amended commit...)

Reply:
lol I just realized that I used NotNil instead of Nil. Sorry about that! 👍